crates/sui-json/src/tests.rs

```diff
@@ -928,3 +928,80 @@ fn test_string_vec_df_name_child_id_eq() {
         child_id.to_string()
     );
 }
+
+#[test]
+fn test_type_tag_to_primitive_layout() {
```

Comment:
We should have som tests for the `primitive_type` function,  (substituting type parameters), and some tests with some more deeply nested type parameters (like `vector<Option<T>>` etc).